src/main/java/Parser.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * The Parser class is responsible for parsing various types of Strings encountered within the program.
+ */
+public class Parser {
+
+    private Task parseTaskToAdd(String consoleArg) throws EmptyTaskDescriptionException, InvalidTaskTimeException,
+            UnknownCommandException, InvalidInputException {
+
+        if (consoleArg.length() <= 4) {
+            throw new UnknownCommandException("Error: Missing task type keyword");
+        }
+        String[] parsedArr = consoleArg.substring(4).split(" ");
+        String keyword = parsedArr[0];
+        if (keyword.equals("todo")) {
+            String name = "";
+
+            for (int i = 1; i < parsedArr.length; i++) {
+                if (i != parsedArr.length - 1) {
+                    name += parsedArr[i] + " ";
+                } else {
+                    name += parsedArr[i];
+                }
+            }
+            if (name.equals("")) {
+                throw new EmptyTaskDescriptionException("Error: Empty task name not allowed");
+            }
+
+            return new Todo(name, false);
+
+        } else if (keyword.equals("deadline")) {
+            String name = "";
+            String time = "";
+            boolean nameOrTime = true;
+            for (int i = 1; i < parsedArr.length; i++) {
+                if (nameOrTime) {
+                    if (parsedArr[i].contains("/by")) {
+                        if (name.equals("")) {
+                            throw new EmptyTaskDescriptionException("Error: Empty task name not allowed");
+                        }
+                        nameOrTime = false;
+                        continue;
+                    } else {
+                        name += parsedArr[i] + " ";
+                    }
+                } else {
+                    if (i != parsedArr.length - 1) {
+                        time += parsedArr[i] + " ";
+                    } else {
+                        time += parsedArr[i];
+                    }
+                }
+            }
```

Comment:
This area looks a little confusing, perhaps could try reducing the amount of nesting occuring?